src/main/webapp/script.js

```diff
@@ -289,52 +289,57 @@ class AutocompleteDirectionsHandler {
 
                 // AQI is only available for routes passing through the grid exactly. 
                 // Or else no data available and will have error saying it is undefined (because it doesnt exist). 
-                // If the mapRow from getGridIndex of stepAQIdoesnt doesn't exist in the data grid, stepAQI is 0 for now. 
-                let stepAQI = 0;
+                // Ignore step if the mapRow from getGridIndex of stepAQIdoesnt doesn't exist in the data grid, 
+                // as setting it to 0 will skew data to 0 if passing through an area with very little data. 
+                // So the route score will only be scored based on available data. 
+                let stepAQI;
                 if (!dataGrid[mapRow]) {
-                    console.log("Row map doesnt exist and is undefined.");
+                    console.log("Row map doesnt exist and is undefined. Ignore this step, skip over iteration.");
+                    continue;
                 } else if (!dataGrid[mapRow][mapCol]) {
-                    console.log("Step aqi dosnt exist because mapCol doesn't exist in mapRow. So cant find and set stepAQI");
+                    console.log("Step aqi dosnt exist because mapCol doesn't exist in mapRow. Skip over iteration.");
+                    continue;
                 } else {
```

Comment:
This last else doesn't need to be an else block; if either other condition is true, the flow of execution leaves the block.

Reply:
But isnt there also the condition where if the other condition isnt true, this could still happen? there are instanes where the mapRow existed, but the mapCol didnt, so stepAQI was still undefined.

Reply:
A simplified version of this loop, as I read it:
```
for ... {
  if (a) {
    doA();
    continue;
  } else if (b) {
    doB();
    continue;
  } else {
    doC();
  }
  doD();
}
```
Where either of the first two conditions are met, we move on to the next iteration of the `for` loop.
If neither of the first two conditions are met, we execute the final `else` condition, and also continue through the remainder of the loop block.
In terms of the flow control, the final `else` is executed only under the same conditions as the rest of the loop, so it's easier to read if the last `else` is removed.

This wasn't true before this PR, but because of the added `continue` to each earlier condition, the flow has changed.

Reply:
oh yes, you're right on the last else block, apologies! I read your comment pointing out to the highlighted "else if" statement instead, and what was what I was referring to.

Reply:
There is a function in the master branch, `getAQI(data, gridIndex)` that will get the aqi and return -1 if it doesn't exist. It's not very useful in this PR because it was merged after this PR started, but it could make this part of the coed simpler